packages/curve-ui-kit/src/shared/ui/DataTable.tsx

```diff
@@ -0,0 +1,155 @@
+import { useRef } from 'react'
+import Typography from '@mui/material/Typography'
+import TableHead from '@mui/material/TableHead'
+import TableCell from '@mui/material/TableCell'
+import TableBody from '@mui/material/TableBody'
+import Table from '@mui/material/Table'
+import TableRow from '@mui/material/TableRow'
+import { SizesAndSpaces } from '../../themes/design/1_sizes_spaces'
+import { useIntersectionObserver } from 'ui'
+import {
+  Cell,
+  Column,
+  ColumnDef,
+  flexRender,
+  getCoreRowModel,
+  getSortedRowModel,
+  Header,
+  Row,
+  SortingState,
+  useReactTable,
+} from '@tanstack/react-table'
+import { useSortFromQueryString } from '../../hooks/useSortFromQueryString'
+import { ArrowDownIcon } from '../icons/ArrowIcon'
+
+const { Sizing, Spacing, MinWidth } = SizesAndSpaces
+
+const getAlignment = <T extends any>({ columnDef }: Column<T>) => (columnDef.meta?.type == 'numeric' ? 'right' : 'left')
```

Comment:
`const getAlignment = ({ columnDef }: Column<any>) => (columnDef.meta?.type == 'numeric' ? 'right' : 'left')`

Reply:
Actually unknown or never should be used not any

Reply:
I tried both `never` and `unknown` but it opens up another can of worms sadly

Reply:
I think the correct is `<T extends unknown>({ columnDef }: Column<T>)` but I kind of agree `({ columnDef }: Column<any>)` is simpler

Reply:
I'd still like to argue for the more complex signature because any will just disable any type checks. Unknown forces us to check when using the values